__init__.py

```diff
@@ -236,29 +236,37 @@ def on_volume_unduck(self, message):
     def set_hardware_volume(self, pct):
         """ Set the volume on hardware (which supports levels 0-63).
 
+            Since the amplifier is quite powerful the range is limited to
+            0 - 30.
+
             Arguments:
-                pct (int): audio volume (0.0 - 1.0).
+                pct (float): audio volume (0.0 - 1.0).
         """
-        self.log.debug('Setting hardware volume to: {}'.format(pct))
+        vol = int(20 * pct + 10) if pct >= 0.01 else 0
+        self.log.debug('Setting hardware volume to: {} ({})'.format(pct, vol))
+
         try:
             subprocess.call(['/usr/sbin/i2cset',
                              '-y',                 # force a write
                              '1',                  # i2c bus number
                              '0x4b',               # stereo amp device address
-                             str(int(15 * pct) + 15)])  # volume level, 0-63
+                             str(vol)])  # volume level, 0-30
         except Exception as e:
             self.log.error('Couldn\'t set volume. ({})'.format(e))
 
     def get_hardware_volume(self):
-        # Get the volume from hardware
+        """ Get the volume from hardware
+
+            Returns: (float) 0.0 - 1.0 "percentage"
+        """
         try:
             vol = subprocess.check_output(['/usr/sbin/i2cget', '-y',
                                            '1', '0x4b'])
             # Convert the returned hex value from i2cget
             i = int(vol, 16)
-            i = 0 if i < 0 else i
-            i = 63 if i > 63 else i
-            self.volume = i / 63.0
+            i = max(i, 0)
+            i = min(i, 63)
+            self.volume = min(max((i - 10) / 20.0, 0), 1.0)
```

Comment:
Clip func could be nice here:

```
vol = clip(vol, 0, 63)
pct = (vol - 10) / 20.0
self.volume = clip(pct, 0.0, 1.0)
```

Reply:
What? is `min(max(x, min_val), max_val)` not clear?

You're definitely right, adding that.